tests/e2e/utils/ack_bootstrap/ack_deployment.py

```diff
@@ -0,0 +1,142 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import json
+import boto3
+
+from e2e.utils.utils import (
+    load_json_file,
+    get_iam_client,
+    get_eks_client,
+)
+from e2e.fixtures.cluster import (
+    associate_iam_oidc_provider,
+)
+
+from e2e.utils.aws.iam import IAMPolicy
+from e2e.utils.ack_bootstrap import common
+from e2e.utils.utils import print_banner, load_yaml_file, write_yaml_file
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def profile_trust_policy(
+    cluster, region, account_id
+):
+    eks_client = get_eks_client(region=region)
+
+    resp = eks_client.describe_cluster(name=cluster)
+    oidc_url = resp["cluster"]["identity"]["oidc"]["issuer"].split("https://")[1]
+
+    trust_policy = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": f"arn:aws:iam::{account_id}:oidc-provider/{oidc_url}"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringEquals": {
+                        f"{oidc_url}:aud": "sts.amazonaws.com",
+                        f"{oidc_url}:sub": [ 
+                            "system:serviceaccount:ack-system:ack-sagemaker-controller",
+                            "system:serviceaccount:ack-system:ack-applicationautoscaling-controller"
```

Comment:
nit: if we just add the sagemaker controller sa, which should be the only one we need, then we can use eksctl to create the sagemaker controller irsa

we can add this in a follow up pr

Reply:
yep good suggestion
will create a common function which uses eksctl in a follow up pr